File: lcof2/剑指 Offer II 038. 每日温度/Solution2.ts
function solve(temperatures: number[]): number[] {
  const ret: number[] = [];

  for (let i = 0; i < temperatures.length; i++) {
    let j = i + 1;
    let n = 0;
    while (j < temperatures.length) {
      n++
      if (temperatures[j] > temperatures[i]) {
        break
      }
      if (j == temperatures.length - 1) {
        n = 0
      }
    }
    ret.push(n);
  }

  return ret;
}

File: lcof2/剑指 Offer II 038. 每日温度/Solution2.py
from typing import List


def solve(temperatures: List[int]) -> List[int]:
  ret = []
  for i, temp in enumerate(temperatures):
    j = i + 1
    n = 0
    while j < len(temperatures):
      n += 1
      if temperatures[j] > temp:
        break
      if j == len(temperatures) - 1:
        n = 0
    ret.append(n)

  return ret
